codegen/plugin.go

```diff
@@ -42,3 +101,10 @@ func RunPlugins(cmd, genpkg string, roots []eval.Root, genfiles []*File) ([]*Fil
 	}
 	return genfiles, nil
 }
+
+// insertPlugin inserts the plugin into the list of plugins at the given index.
+func insertPlugin(p *plugin, index int) {
+	plugins = append(plugins, &plugin{})
```

Comment:
Do we really need a function here? seems it should be possible to do:
```go
plugins = append(plugins[:i], append(p, plugins[i:]...)...)
```

Reply:
Ah nice.. Just now found that in https://github.com/golang/go/wiki/SliceTricks